sim/sim.go

```diff
@@ -107,6 +104,10 @@ func (s *Simulation) Run(instanceCount uint64, maxRounds uint64) error {
 				}
 			}
 
+			// Instantiate the next instance even if it goes beyond finalInstance.
+			// The last incomplete instance is used for testing assertions.
+			currentInstance = s.ec.BeginInstance(*decidedChain, pt)
```

Comment:
This was the cause of much head scratching, where: `StartInstance` was being called for the instance that just started instead of the one that just finished. This would then resulted in participants deciding on the wrong base. 

Hence, begin next instance after lagging participants are notified of the completion of current one.